component/src/main/java/org/wso2/extension/siddhi/store/influxdb/InfluxDBTableUtils.java

```diff
@@ -0,0 +1,108 @@
+ */
+
+public class InfluxDBTableUtils {
+
+    private InfluxDBTableUtils() {
+        //preventing initialization
+    }
+
+    public static boolean isEmpty(String field) {
+
+        return (field == null || field.trim().length() == 0);
+    }
+
+    /**
+     * utility method to map tag values  to attributes before write to database;
+     */
+    public static Map<String, String> mapTagValuesToAttributes(Object[] record, List<String> attributeNames,
+                                                               List<Integer> tagPositions) {
+
+        Map<String, String> attributesValuesMap = new HashMap<>();
+        for (int i = 0; i < record.length; i++) {
+            for (int j = 0; j < tagPositions.size(); j++) {
+                if (i == tagPositions.get(j)) {
+                    attributesValuesMap.put(attributeNames.get(i), record[i].toString());
+                }
+            }
+        }
+        return attributesValuesMap;
+    }
+
+    /**
+     * Method to map filed values to attributes before write to database.
+     *
+     * @param record         Object array of runtime values
+     * @param attributeNames list of attribute names
+     * @param tagPositions   list containing position of tag keys
+     * @return
+     */
+    public static Map<String, Object> mapFieldValuesToAttributes(Object[] record, List<String> attributeNames,
+                                                                 List<Integer> tagPositions) {
+
+        Map<String, Object> attributesValuesMap = new HashMap<>();
+        int count;
+        for (int i = 0; i < record.length; i++) {
+            count = 0;
+            for (int j = 0; j < tagPositions.size(); j++) {
+                if (i != tagPositions.get(j)) {
+                    count++;
+                }
+            }
+            if (count == tagPositions.size()) {
+                attributesValuesMap.put(attributeNames.get(i), record[i]);
+            }
+        }
+        return attributesValuesMap;
+    }
+
+    /**
+     * Utility method to map time value to attributes.
+     *
+     * @param record        Object array of runtime values
+     * @param tableName     name of the table
+     * @param timePositions contains position of time attribute
+     * @return
+     */
+    public static long mapTimeToAttributeValue(Object[] record, String tableName,
+                                               int timePositions) {
+
+        String time = "";
+        try {
+            for (int i = 0; i < record.length; i++) {
+                if (i == timePositions) {
+                    time = record[i].toString();
+                }
+            }
+            long times = Long.parseLong(time);
+            return times;
+        } catch (NumberFormatException e) {
+            throw new InfluxDBTableException("Specify 'time' attribute in type Long in the table definition "
```

Comment:
Seems 'time' attribute is mandatory? This should be validated at init level

Reply:
https://github.com/wso2-extensions/siddhi-store-influxdb/pull/1/commits/520b093086a08e2d8d9df4297dcd71914e15813b